components/net_traits/image_cache_task.rs

```diff
@@ -86,21 +86,15 @@ pub fn load_image_data(url: Url, resource_task: ResourceTask, placeholder: &[u8]
                 return Ok(image_data);
             }
             ProgressMsg::Done(Err(..)) => {
-                // Failure to load the requested image will return the
-                // placeholder instead. In case it failed to load at init(),
-                // we still recover and return Err() but nothing will be drawn.
-                if placeholder.len() != 0 {
-                    debug!("image_cache_task: failed to load {:?}, use placeholder instead.", url);
-                    // Clean in case there was an error after started loading the image.
-                    image_data.clear();
-                    image_data.push_all(&placeholder);
-                    return Ok(image_data);
-                } else {
-                    debug!("image_cache_task: invalid placeholder.");
-                    return Err(());
-                }
+                return Err(());
             }
         }
     }
 }
 
```

Comment:
Perhaps this one could be defined in the same position in source file as LoadPlaceholder (since both enums are used to control the behavior)? While at it, I would add an one liner comment explaining the use of each enum.